presto-main/src/test/java/io/prestosql/sql/planner/TestDynamicFilter.java

```diff
@@ -216,4 +216,29 @@ public void testSubTreeJoinDFOnBuildSide()
                                 exchange(
                                         project(tableScan("part", ImmutableMap.of("PART_PK", "partkey")))))));
     }
+
+    @Test
+    public void testNestedDynamicFiltersRemoval()
+    {
+        assertPlan("WITH t AS (SELECT o.clerk FROM (" +
+                        "(orders o LEFT JOIN orders o1 ON o1.clerk = o.clerk) LEFT JOIN orders o2 ON o2.clerk = o1.clerk)" +
+                        ") SELECT t.clerk FROM (orders o3 JOIN t ON t.clerk = o3.clerk)",
+                anyTree(
+                        join(INNER, ImmutableList.of(equiJoinClause("ORDERS_CK", "ORDERS_CK6")),
+                                anyTree(
+                                        node(FilterNode.class,
```

Comment:
what is filter condition here?

Reply:
updated to assert on the expected dynamic filter